### ISSUE TITLE
Proposal: SwiftBar + BitBar joining forces

### ISSUE THREAD

**Issue by matryer** [swiftbar/SwiftBar#95]
SwiftBar is everything I dreamed BitBar could become. It isn't written in Objective-C for a start; what's not to like? Over the years, the BitBar codebase became stale, difficult to maintain, impossible to build in some cases, and a frustration to many.

**I would like to propose that we take the best of both projects, and join forces.**

SwiftBar brings:

* A fresh codebase written in a modern language 🎉 
* This makes it much easier to contribute to, bug fix, and maintain
* Swift is accessible to many more developers too, including more junior devs who would be able to get involved (a point well made by @melonamin)
* A built-in plugin browser, which is a massive boost in UX, and opens the project up to non-techy users

BitBar has:

* History and reputation (it's used by lots of people, and has been featured in a few places, like [TechRadar](https://www.techradar.com/uk/how-to/how-to-power-up-your-mac-with-bitbar) and [Mac Format magazine](https://www.dropbox.com/s/cvfv53vt73m5lua/Mat%20Ryer%20BitBar.jpg?dl=1))
* Momentum (more and more people are still discovering BitBar every day)

## How?

This is how I think we should do it:

* Use the BitBar name
* PR the SwiftBar code into the BitBar repo (Ideally we would keep the history of **all** contributors)
* Find a neutral org on GitHub and _move_ the BitBar repo there (unfortunately `bitbar` is taken, but I'm sure we can find something that works). Moving the repo is important so we get the GitHub redirection from the current repo
* Agree some product ownership and decision making processes to ensure the project is able to continue to develop into the future (and more importantly, doesn't go stale!)

Whatever we decide, I'd like to thank @melonamin and everyone who contributed to SwiftBar for building it.

**Comment by yanzhang0219** [comment]
Even though BitBar is hard to be maintained now (because the author has no much time, no new maintainers, etc), why not achieve it and declare that SwiftBar is the successor in its README? BitBar users can switch to use SwiftBar without many efforts because one of SwiftBar's original points is to support BitBar's API and its existing plugins.

As you said, if merged and still use BitBar name, I believe @melonamin will still be responsible for most of the follow-up development. I say this because you have been looking for a maintainer for BitBar for a really long time (since 2016?), however, so far there is still no one standing up, thus after merging, I think the only active developer will still be @melonamin . So, it is unfair to @melonamin. One person develops an amazing project, and then his code and his repo is merged into another, and finally, he needs to continue to contribute efforts for continuous development. Emmmm.. It's totally unfair (just in my opinion at least). In the future, no one will remember there is a repo called SwiftBar in the world, no one will know there was a period of deathly stillness for BitBar (almost 4 years of silence), the only thing that people could praise is BitBar.

So my suggestion is to keep SwiftBar.

**Comment by tjluoma** [comment]
I also think that it would be better to keep the projects separate.

**Comment by matryer** [comment]
Hey @yanzhang0219, thanks for sharing your view.

Archiving BitBar and pointing to SwiftBar is indeed another option, but it doesn't take advantage of the brand awareness that BitBar has already built up.

I share your concern about crediting contributors. Perhaps where we differ is that I apply that same argument to everybody who has contributed to BitBar over the years. There's much more to a project like this than the code, consider the ideas, features, design, etc. which SwiftBar inherits and builds upon.

I was also concerned with making sure it was fair to @melonamin and we had a great conversation today. He is happy to proceed with the plan I outlined. I was particularly impressed by his collaborative attitude; he really cares about what's best for the project in the truest spirit of OSS.

To your point about the history of this, I love the idea of adding a short history to the README so that nobody's contributions are forgotten, highlighting @melonamin in particular for unblocking things.

**Comment by matryer** [comment]
@tjluoma thanks for sharing your view. I'm surprised to hear that, perhaps you could elaborate a little? In particular, can you answer the specific points I have made?

**Comment by yanzhang0219** [comment]
Actually my desire is keep the author of the project to be @melonamin, because he is so active and his work is so organized. I used BitBar before, when I got an issue, no one replied me. I have to beg for a answer or solution. I hate that experience. On the opposite, after I switched to SwiftBar, @melonamin is very very responsive, trying to answer every question. Also, by looking at how many open issues remaining we can noticed that he is so well organize the repo, the milestone list shows what will be released next. I really like and enjoy this experience.

After merged, @melonamin will become a normal contributor or maintainer, probably he will lost his enthusiasm to do so much.

**Comment by matryer** [comment]
@yanzhang0219 it sounds like we're both fans of his :)

**Comment by yanzhang0219** [comment]
We can just re-examine the BitBar repo and take a look at how many open issues vs close ones, 163 vs. 233, and most importantly, most of these open issues still have no solutions so far. I remember there is an issue instructing us how to build this repo on our own if we want a bug free version. Also, we can glance at its commits page, since 2016, that repo has not been maintained well. I understand that once a project gets mature, its development will become slow, this is normal. However, leaving many bugs there is not a good choice. 

I really dislike that experience. SwiftBar, as lease for me, is a daily must-have. Especially, when I encountered many bugs and issues when using BitBar and no one helped me, SwiftBar is my last resort and saves me. 

I wish SwiftBar could live longer, prosperous all the time.

**Comment by matryer** [comment]
I think the point about the BitBar repo going stale has been well made. Nobody is disagreeing with you @yanzhang0219.

**Comment by yanzhang0219** [comment]
Just a few words I want to say. BitBar benefit from its creative idea, so it got many users. And just because these users, BitBar harvest those reputations (mentioned in the first post above). So I think the bottom line should be at least providing an available version for most of the users. I know the author is quite busy, so we don't ask for any new features, but just breakdown bugs(e.g., issues caused by new OS released) fix to make sure there is an available one there. After acquiring many reputations, BitBar abandons its users.

The author had no time, I know, but other contributors has already fix some bugs (e.g., that terrible text vertical alignment bug), the author just need to log in his github, and merge it. I think the process may take 5 min? He is so busy and cannot share that 5 min? He didn't merge it, so some contributors have to ask users who had this issue to use their forked repo until there is an official merge. Then many months or years past.. terrible.

**Comment by matryer** [comment]
@yanzhang0219 Thanks for your continuous and unending criticism.

**Comment by matryer** [comment]
@yanzhang0219 I appreciate your passion on this subject, but do you have anything constructive to add?

**Comment by tjluoma** [comment]
My only additional comment is that I see no reason to ask @melonamin to do any of the additional work which would be necessary for this, not to mention likely be seen as a "takeover" by some BitBar fans (and possibly the BitBar developer), and potentially sour the reputation of SwiftBar as "the project that tried to force a takeover of BitBar" and leave everyone with a bad taste in their mouth.

In a perfect world, this suggestion might make sense.

In the actual Internet that exists in 2020, I think it is a fraught idea with very little potential upside and a whole lot of potential downsides.

Given the "meh, build your own if you need fixes" attitude of the BitBar developer towards people who have submitted fixes or asked for feature for the past four years, I'd just as soon stay clear of anything that he might feel some ownership of. A clean new version gives us a clean break from all of that.

**Comment by yanzhang0219** [comment]
Thank you. My constructive advice is:
**Don't do anything. Keep SwiftBar.** 
I think more and more people will find SwiftBar. Actually, I think SwiftBar's functions are very complete, just some bug fixes needed.

**Comment by matryer** [comment]
@melonamin Perhaps you could share your thoughts?

**Comment by matryer** [comment]
@yanzhang0219 thank you so much for saying that BitBar was a creative idea. That means a lot.

**Comment by bilal-fazlani** [comment]
I love this idea! 

Users bet to keep using bitbar and bitbar gets awesome upgrades! 

An existing userbase of bitbar is huge and can really benefit from active development.

**Comment by bilal-fazlani** [comment]
If this happens,

I wonder if people will be able to do "brew upgrade bitbar" and get the new swiftbar version?

**Comment by yanzhang0219** [comment]
Userbase is huge, then already silent for 4 years without maintenance. 
Then after these two merged, why "benefit from active development"?

**Comment by matryer** [comment]
@bilal-fazlani Yes, if they merged, BitBar would get updated and the SwiftBar code would take over, so `brew update bitbar` would download the latest version.

**Comment by bilal-fazlani** [comment]
Bitbar and Swiftbar are both awesome. 
About some of above criticism of bitbar, my views are a person created something unique which reached thousands of people and people loved it. 

The project was maintained for long time but then as an open source developer myself, I can understand how it can get difficult to keep maintaining a software project for years and without any monetory benefits. 

All the users who appreciate and also users who criticize bitbar have used and benefitted from awesome work of the author without really sponsoring it. And the author doesnt really owe it it to us to resolve the github issues we throw at him. Its open source and its free. Open source software means just open source software. Doesnt mean it comes with a lifetime support for free. The benefit it provides is a building block on which you can build your own project if you want. 

I think its wonderful thought if both the authors think their combined efforts can have even larger impact for users in future.

**Comment by matryer** [comment]
@bilal-fazlani Thanks for sharing your insights. The criticisms of BitBar going stale are valid, which is why it's so surprising my efforts to address that are being met with such hostility. 🤷 My focus (and that of @melonamin from what I gather from our conversation today) is about what's best for the project.

Of course, the reality of what was going on with BitBar is much more complicated than has been expressed in this thread.

**Comment by yanzhang0219** [comment]
If finally they are merged, I wish @melonamin could the owner of the merged repo. The style of how he is managing this repo is fantastic. He makes everything clear to all users. He marks and categories each issue. He uses the milestone as a roadmap to keep tracking of the development process. His release changelog is great as well. 

The author said he has no time to maintain BitBar 4 years ago, will he have time after merging?

> All the users who appreciate and also users who criticize bitbar have used and benefitted from awesome work of the author without really sponsoring it. And the author doesnt really owe it it to us to resolve the github issues we throw at him. Its open source and its free. Open source software means just open source software. Doesnt mean it comes with a lifetime support for free. The benefit it provides is a building block on which you can build your own project if you want.

But at least, he should log in to his GitHub to merge the bug fixes, so that 13.8K users can continue to use it. He doesn't need to fix anything if he doesn't have time, but no others can do the merging stuff.

**Comment by matryer** [comment]
The five maintainers of the BitBar project got blocked for legitimate reasons. Primarily it was becuase it's written in Objective-C, and the operating system, Xcode, etc. were changing a lot. The app worked the whole time, the build toolchain was where the trouble was. This is why it needed a rewrite in Swift, and this is what I was trying to get done (without knowing Swift myself).

Some people don't understand this, and I understand that, but I hope we can take some of the heat out of this thread and get back to being constructive to see if we can find a way forward.

**Comment by yanzhang0219** [comment]
It's README doesn't mention anything like the maintainers are blocked, no one knows that. Also, even though they are blocked, the author is not blocked, and he should deal with those pending merging request. Many of them says like "please merge", but 4 years goes away. The author also could open an issues saying like "our maintainers are blocked , or xxx", but we don't see something like this. 

Okay. I'll stop here. 

Last but not least, I want to say to every user of SwiftBar, if you guys think SwiftBar is great and shouldn't die, let it there, or make @melonamin as the new owner, otherwise SwiftBar's future will be gloomy. 

Thanks.

**Comment by tjluoma** [comment]
The idea in this proposal has been raised. It does not need to be accepted or agreed to by anyone for the project to continue. If @melonamin wants to pursue it, fine. Everyone else should happily accept whatever the person actually doing the work wants to do. If that's merging, fine. If it's not, fine. 

My objections are based on the desire to avoid this project getting burdened with the baggage of another that has an extremely cloudy history (including parts that people aren't really sure what happened and some people may or may not be blocked?) and has been essentially unmaintained for 4 years. Dumping that onto a new project does not seem like a good idea for the overall health and future of the project.

As a user of iTerm2, I can tell you that I have never once looked to see what happened to iTerm1. I'm just happy to use iTerm2. If SwiftBar ends up as the next generation of BitBar, I don't think it will necessarily be harmed by the fact that it isn't the same repo name as the original. People who knew the old name will learn the new name. Most people never knew of the old name and won't need to.

**Comment by melonamin** [comment]
Oh boy… 😅

@matryer, sorry I left you here all alone, I had a crazy busy day.

To **All**,

First and foremost, let me share some background with you:

* The week I’ve released SwiftBar I reached out to @matryer on Twitter to let him know. Partly because SwiftBar uses bitbar-plugins in “Plugin Repository” and partly because I was a BitBar user for a long time and wanted to let him know that the idea continues to live in other forms
* @matryer suggested to have a call and chat about how we can partner to make the best product for the users
* After our chat I suggested @matryer to start this discussion

So he is a welcome guest here, **please treat him this way**.

Secondly, all successful open source projects are a product of the work of many people, for BitBar it is a combination of:

* @matryer, who had this idea and created the original app
* A few other developers who helped BitBar to evolve
* Hundreds of people who built all sort of plugins
* People who spread the word in social media and other places

For SwiftBar:
* BitBar itself, we’re building from a well-formed (and loved) idea, we are standing on BitBar shoulders.
* All contributors to this repo(issues, code, discussions) that I’m lucky to have
* People who spread the word in social media and other places

So in my mind, a good open-source project has not only a “founder\owner” but a group of contributors who move the project forward together. This is by the way one of the reasons why I’ve created a separate organization for this repo.

I don’t think there is a lot a value in trying to find who to blame for a slowdown of BitBar development, so please don’t, what we should focus on is - how to make the best product? Best meaning - available to all users, featureful, supported & maintained.

I don't have a formed opinion at this moment, because I'm the first time in this position, so I retreat to the wisdom of crowds(and I'm already getting new inputs from this thread).


I see three roads from where we are:

####  1. Merge them, in this context it means something like @matryer  described above
**Pros:**
* we can make more users will be "happier" faster
* we preserve project history, honoring the contribution of other people
* we keep using a well-known name (I'm not strong in marketing and not sure if it is valuable or not, and http://bitbar.com concerns me too...)

**Cons:**
* quite a lot of work dedicated to merging 
* risk that @matryer  and I will hate each other and kill the project (after talking to him don't see it happening, he is a nice guy)

####  2. Leave as it is, BitBar and SwiftBar continue on their own
Not much will change from my point of view, this is how this project started and was intended to be. 

**Pros:**
* no "new" work
**Cons:**
* potentially confused users
* two groups of people working pretty much on the same stuff

####  3. Deprecate BitBar and tell people to go here

**Pros:**
* SwiftBar will probably get more users
**Cons:**
* people who contributed to BitBar will see that it is "deprecated", nobody likes when it happens
* new "brand"


Once again, I had a long day and not in my sharpest state, so I probably missing something, please peach in your opinions. @tjluoma thoughts and experience were particularly helpful to me in this thread and highlighted some things I didn't think about.

p.s. I summon @cliss and @jasonsnell since they are not only longtime users of BitBar but brought most of the people to SwiftBar (I understand if you ignore this, it is a long read.)

**Comment by jasonsnell** [comment]
Hi everyone.

I don't even remember how I found BitBar, but I first wrote about it [five years ago almost to the day](https://sixcolors.com/post/2015/12/gift-guide-mac-apps/). It's been a must-have part of my Mac setup ever since.

First, I want to say that people should lay off @matryer - we owe him a lot for building a great app and he owes us essentially nothing. I think it's awesome that he's still involved in any way with this, because it would've been easy for him to just walk away. Thank you, Mat.

I also want to thank @melonamin for stepping up and writing SwiftBar, which has really duplicated everything I liked about BitBar in short order, at a time when Big Sur was making BitBar look broken. BitBar was a utility that I felt was locked in place; it was what it was and that's all it was ever going to be. SwiftBar feels like it has momentum and could become a lot better in a short period of time. That's a big deal.

I am not a developer so I can't say what the right approach is. I will say that I am concerned with the collision with BitBar Inc. and the BitBar DevOps service. But I don't really have an attachment to either name.  I also don't know about the ins and outs of keeping the projects separate or merging one into the other. If the swiftbar project became `bitbar2` and it was the official continuation of BitBar, I don't see how that slights anyone who contributed to BitBar, but: **I don't understand GitHub politics**.

I think the most important thing in the long term is to have a community and developer/developers who are enthusiastic about maintaining the project. Not the name, where it's hosted, anything like that. The real threat is that there ends up being two separate projects that diverge and a lot of resources that could've been put toward a single project are diluted.

Thanks for listening to an enthusiastic user.

**Comment by yanzhang0219** [comment]
**The only, unique thing I care about is SwiftBar can be alive for a long time with the **necessary** bug fixes**. "Necessary" means like merging contributor's PRs (I know users are active and they can fix any bugs, but the owner or maintainers need to merge them and release a new version to make it survive). Actually, it is already great if no new features added in the future.

So, from this point of view, **the safest way is 2 and 3.** 

> what we should focus on is - how to make the best product? Best meaning - available to all users, featureful, supported & maintained.

Exactly. These words "available", "supported", "maintained" should belong to SwiftBar. If it is merged into BitBar, I don't believe these words could retain for a long time. 

> we can make more users will be "happier" faster

Not merging can still get more users. Especially when BitBar users notice there is a repo called SwfitBar and give it a shot, I guarantee they will switch to SwiftBar. Also, among the 13.8k BitBar users, the active users probably are not too many, and these active users will find SwiftBar very quickly.

> we preserve project history, honoring the contribution of other people

As I said, SwfitBar can make its own history and honors.

> two groups of people working pretty much on the same stuff

This cannot happen. The author of BitBar is too busy to maintain. Other ones will switch to SwiftBar to make contributions to SwfitBar. Actually, no one is working on BitBar, as I said in my above posts (no one solves issues, no one merges bug fixes, a little alignment bug can be there for 2-3 years, unbelievable). 

I prefer choice 2. Keep both and make users decide which one they are going to use.

Thank you.

**Comment by matryer** [comment]
Thanks @melonamin and @jasonsnell - I found it very refreshing reading your comments. I have no ego about BitBar, for me it's about what's best for the project too. I'd love to hear from some other voices if we can make space for them.

**Comment by tjluoma** [comment]
@matryer I apologize that I did not realize you were the original creator of BitBar. That was just a lack of information on my part because I was not around when it started. I wish I had been, it was exactly the type of thing that I spent a lot of time looking for.

It seems like perhaps @matryer and @melonamin had already talked about this idea and were bringing it to us for discussion? If that’s the case, it would have been helpful to know. My primary concerns were that this idea was forcing / pushing an idea onto @melonamin that he might not be interested in pursuing.

My intention was not to insult or attack anyone, only to disagree with an idea. If it came across otherwise, I apologize for my mistake. 

FWIW, here’s where I’m coming from (the view from 20,000 feet):

1. Wished for a way to put script info into the menu bar for a long time.

2. Unable to find a way to do that for a long time

3. Eventually _finally_ found [TextBar](https://www.macstories.net/mac/textbar-puts-your-text-into-the-menu-bar/) which I loved, but which never really caught on with the wider community, probably because it’s a paid-app [$5] and (many) nerds are cheap.

4. Learned about BitBar in 2020, sounded great. Moved all of my TextBar scripts over to BitBar.

5. Learned that BitBar development was no longer active, even for bug fixes, big disappointment. Hints that there was some kind of upheaval/unease/etc in the project’s past that I didn’t know about and didn’t want to know about.

6. Wondered again why / frustrated again that TextBar never caught on with more people. Moved all of my scripts back to TextBar.

7. Heard about SwiftBar

8. Excited about SwiftBar

9. “Holy cow! SwiftBar is going gang-busters! How do we keep this going? [Can we contribute financially?](https://github.com/swiftbar/SwiftBar/discussions/92) The last thing I would want would be for something to happen that would result in SwiftBar not continuing along it’s so-far-awesome path of development!

10. Then I read “Hey, what if we tried to merge BitBar and SwiftBar?” and my brain went “DANGER WILL ROBINSON! DANGER!”

11. Jason said that he doesn’t understand GitHub politics, and I don’t claim to know them that well either. However, my somewhat-considerable experience with people in my professional life has taught me this: “Unresolved issues, especially dealing with things like anger, hurt, and resentment, have a way of lying dormant for a while until something happens and then ***watch out*** because here it comes.” It can be a hugely destructive force that ends up being a huge time-suck and creates a lot of emotional damage until it is properly resolved. I don't know (or want to, really) what happened in BitBar's past, but I also had this feeling on the back of my neck that it was not a place that I wanted to spend a lot of time. Felt like wading into very murky, unclear water… and, as someone who lived in Florida for a while, that's just not something you ever want to do, because who knows what is beneath the surface, but there's probably something, and it probably bites (at least).

12. My experience with Nerds on the Internet is that they can be amazingly awesome and great, or some of the absolute worst.

Take those 12 points together and you can probably better understand my responses above. Not saying they're right or wrong, just by way of explanation or clarification.

I very much wish that I could contribute something tangible to help this project survive and thrive, but unfortunately, I can't code beyond shell scripting.

**Comment by cliss** [comment]
Hi, everyone. Enthusiastic user here and occasional [poor] open-source developer/maintainer.

I echo what @jasonsnell said: first and foremost, incredible appreciation should be shown to both @matryer and @melonamin for their incredible work.

I, for one, did not **at all** read Mat's idea about replacing Bitbar with SwiftBar as ego-driven. I totally understand the perks of doing so:

* Existing users get a new shiny with almost no work on their part
* Years of press/commentary about Bitbar will point to the latest and greatest implementation of this idea, which so happens, today, to be Swiftbar
* Nothing else necessarily has to change; on paper, it's all upside

So, as Jason said, I think everyone needs to pump the brakes on the attacks — implied or otherwise — on Mat.

With regard to Alex's three paths forward, I can make vociferous defenses of any of them. Given Alex is likely the one to take on the work, and that this is his project for which he owes us nothing, I think we should follow his lead. I, for one, would not begrudge him **at all** if he values a clean break and/or clear ownership over what is an extremely impressive achievement. Alex did right by Mat, in my eyes, by making it clear where the inspiration for his work came from.

I don't personally see the need to cater to "legacy" Bitbar users. This is a power-user app that has power-user intricacies. Those that relied on Bitbar will surely have found Swiftbar. Those who were just passive/casual users will not. 🤷 

<hr />

With all of that said, here comes my completely emotional and logic-Liss opinion: why fuck with what works? Keep everything here at Swiftbar, and move on with life. Sometimes a clean break is nice, and the lack of "namespace pollution" with other flavors of "Bitbar" (like the DevOps thing) is an added benefit. Less distraction from the *actual* goal of making the best app possible is a win in my book.

Again, I took it as a very mature and kind thing of Mat to offer the "keys to the kingdom" to this project; I don't think it was ego. Personally, these implementation details are mostly just nerd drama. All I care about is that my scripts show up in the menu bar, and that Alex isn't [quickly] led to the path all open-source maintainers end up walking down: the one to retirement.

-Tact Liss

**Comment by yanzhang0219** [comment]
I almost have the similar path with @tjluoma. I strongly agree with the 11th point. Especially this sentence:

> Jason said that he doesn’t understand GitHub politics, and I don’t claim to know them that well either. However, my somewhat-considerable experience with people in my professional life has taught me this: “Unresolved issues, especially dealing with things like anger, hurt, and resentment, have a way of lying dormant for a while until something happens and then watch out because here it comes.” It can be a hugely destructive force that ends up being a huge time-suck and creates a lot of emotional damage until it is properly resolved. I don't know (or want to, really) what happened in BitBar's past, but I also had this feeling on the back of my neck that it was not a place that I wanted to spend a lot of time. Felt like wading into very murky, unclear water… and, as someone who lived in Florida for a while, that's just not something you ever want to do, because who knows what is beneath the surface, but there's probably something, and it probably bites (at least).

**Felt like wading into very murky, unclear water.** Lol. This is really an appropriate metaphor. Just go back to BitBar's repo to take a look at how many opened, unorganized, unmaintained issues there. SwiftBar will benefit nothing from the merging but lots of dirty works.

**Comment by matryer** [comment]
I am going to move this conversation away from this thread due to the repetitive toxicity of some of the commentators.

**Comment by yanzhang0219** [comment]
Interesting. Actually you are not facing up the problem but running away from them. Look at this (https://github.com/matryer/bitbar/pulls), how many pending, unresolved PRs there? 

Most of them are not aimed at adding new features, but the desperate bug fixes to ensure BitBar can work continuously. However, you were gone and didn't resolve them. You even didn't say anything in the PRs' conversation, and the users there were so frustrating and helpless. Most of the bug-fixing PRs were opened there in 2016.

**I know BitBar is an open-source project, but when it has a large userbase, a sense of responsibility will be reflected. Yes, you don't own anything to the users theoretically. But emotionally, you should do something for your users, after all, you got the reputation from the users, right?**

If SwiftBar is merged into BitBar, SwiftBar, a fantastic project, will be pulled into this murky water. SwiftBar will be ruined. Everyone can imagine what happens in the future. This is as if you move a green, flourishing tree onto a grassy place overgrown with weeds, leaves. What will happen eventually, this tree will be invisible, submerged by the messy.

**Comment by matryer** [comment]
I'm sorry for shutting down this conversation; I regret that. I wasn't having the best day and that leaked out. Thanks to @melonamin for giving me the opportunity to apologise.

Additionally, I did not intend to dismiss the frustration expressed in this thread, however much I wish it could be expressed differently. I share a lot of it. I'm not taking the criticism personally, especially since I haven't been a maintainer in the repo for some years, but I would like to take a minute to defend the BitBar maintainers. In my experience they are genuinely lovely people, who have only ever done their upmost to help, in the best tradition of OSS. Having said that, I apologise to anybody who was made to feel unwelcome or unheard in that repo. I know what that's like from this experience, and it's not pleasant.

Some really solid points have been made along the way. Specifically the BitBar name colliding with other projects, the advantages of a fresh repo, and a fresh owner, the name a few.

**How about we let this idea cool down a bit and revisit it in the new year?**

Finally, if you don't mind me asking, when you reply please make an extra effort to present your views in a positive way, and avoid hammering home points that have already been well made.

**Comment by R0dri** [comment]
Hey, im new to the bitbar/swiftbar comunity so I hope my view can contribute to make a good decision on the topic.

First and foremost, thank you for making this project. I made a plugin last few days and I really love its simplicity and power. A great idea, well made.

New year, new spirit, new plans.

I really think that the way to go is deprecating Bitbar. Either renaming swiftbar repo or creating a new one. If "deprecating" is a strong word, Why not creating a new repo "Bitbar 2" and saying it merged with swiftbar.

But, PLEASE update Bitbar README soon, it was a headache finding swiftbar and trying to figure out what was going on until I found this post.

**Comment by austinbutler** [comment]
> #### 3. Deprecate BitBar and tell people to go here
> **Pros:**
> * SwiftBar will probably get more users
>
> **Cons:**
> * people who contributed to BitBar will see that it is "deprecated", nobody likes when it happens
> * new "brand"

Due to the naming conflict with "BitBar" this is my vote (even though honestly I like that name the best). I knew BitBar had gone a bit stale, so today I went looking to see if it's still the best way to put scripts in your menu bar and ended up finding some DevOps tooling. SwiftBar is uniquely tied to this project.

By setting BitBar repo to read-only after updating the README to point to SwiftBar, you get the best of both worlds. Existing users can keep using SwiftBar (until it breaks, anyways), new and existing BitBar users can easily discover SwiftBar, and there's almost no work required to make it happen.

Everyone please keep in mind, both BitBar and SwiftBar are software created by others in their spare time _for free_! What a deal! So there's no reason to keep belaboring the point that BitBar hasn't been actively maintained and rag on the author (who is here trying to unify the project in the best interests of the users of both projects).

**Comment by melonamin** [comment]
@matryer just [announced](https://github.com/matryer/bitbar/issues/607) BitBar reboot, so I'm locking this issue since further discussion is pointless.

One can say that's it is a shame that in the future, there will be two projects so similar in functionality. I agree that in an ideal world, we all should have one perfect "BitBar" with a happy developer behind it. But the world is not that way, so don't be upset about it.

We'll see, maybe Mat and I will find a way to collaborate and share a plugin repository or something like that.
